### ISSUE TITLE
Debug integration test failure

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#92287]
Retries were added in 5cf50df3be4ea65e01da16e9e6d6e0f674d019d4, this was not enough apparently.

```
  169 passing (1m)
  1 pending
  1 failing
  1) Debug start debugging:
     AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: S {
  _configuration: {
    __pendingTargetId: '8889',
    __workspaceCachePath: '/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/t8732wpx0QdV4rB56/d/data/User/workspaceStorage/1337f4a9/ms-vscode.js-debug-nightly',
    __workspaceFolder: '',
    address: 'localhost',
    disableNetworkCache: true,
    name: '[8889]',
    outFiles: [
      '/Users/runner/runners/2.165.0/work/1/s/extensions/vscode-api-tests/testWorkspace/**/*.js',
      '!**/node_modules/**'
    ],
    outputCapture: 'console',
    pathMa... == undefined
      at Context.<anonymous> (extensions/vscode-api-tests/src/singlefolder-tests/debug.test.ts:42:10)
      at processImmediate (internal/timers.js:439:21)
```

https://monacotools.visualstudio.com/DefaultCollection/Monaco/_build/results?buildId=69274&view=logs&j=a5e52b91-c83f-5429-4a68-c246fc63a4f7&t=087c9a82-9b9b-5b5f-17f2-a070cb4808e3

**Comment by bpasero** [comment]
I really think this test should leverage mock debug and not require a process to start.

**Comment by isidorn** [comment]
The problem was that the re-retry did not clear the previous state.
Now we should be good. Fixed via c91e2b62d7
If it keeps on failing I will look into using mock debug
